lib/portlayer/exec/container.go

```diff
@@ -161,7 +161,16 @@ func newContainer(base *containerBase) *Container {
 		// set state
 		switch base.Runtime.PowerState {
 		case types.VirtualMachinePowerStatePoweredOn:
-			c.state = StateRunning
+			// the containerVM is poweredOn, so set state to starting
+			// then check to see if a start was successful
+			c.state = StateStarting
+			// If any sessions successfully started then set to running
+			for _, s := range base.ExecConfig.Sessions {
+				if s.Started != "" {
+					c.state = StateRunning
```

Comment:
We have `container.updateState` and could potentially call that from `container.refresh` but ok if it doesn't go in this PR.
@cgtexmex please open a follow on issue and tag as tech debt.